ext/B/B.xs

```diff
@@ -1422,6 +1422,30 @@ object_2svref(sv)
     PPCODE:
 	ST(0) = sv_2mortal(newRV(sv));
 	XSRETURN(1);
+
+bool
+is_bool(sv)
+    B::SV   sv
+    CODE:
+    RETVAL = SvIsBOOL(sv) ? TRUE : FALSE;
+    OUTPUT:
+        RETVAL
+
+bool
+is_bool_yes(sv)
```

Comment:
I'm not sure why this is useful (also the `no` below it). If anyone cares to check they can test it from perl-land just as easily

```perl
  if($b_sv->SvIsBOOL && $b_sv->SvTRUE) { ... }
```

Reply:
When you are serializing the information, you need a way to query the B object so it round trips properly in another process.

Reply:
I think this provides a better abstraction, where you could change internal without changing all consumers
if this would change.

Reply:
Yes. But why do you need to add a new XS method on the B::SV class (`is_bool_yes`) which just returns information that is the `and` combination of two other methods that already exist (`IsBOOL and TRUE`)?

Reply:
> I think this provides a better abstraction, where you could change internal without changing all consumers
if this would change.

This isn't about internals. IsBOOL returns true for any boolean. TRUE returns true for any value that perl considers true. Thus the combination `$sv->IsBOOL and $sv->TRUE` is simply the combination of those two facts. It doesn't need a special method.

This would be like adding a special method to ask if a given SV is numerical and the value 1. Why bother? We can just check those two facts, separately, using the regular query methods. No need to special-case it.

Reply:
This is assuming that this will always be the way to check if the boolean is true or false.

I was under the assumption that this could change overtime and we could start using one shared SV for true or false at one point or whatever other optimization we decide in the future.

If we think that `IsBOOL and TRUE` is always going to work, then indeed we can consider removing the `is_bool_yes` and `is_bool_no` helpers. I was trying to provide abstraction for the user which should not track how this is implemented behind.

This would add more boilerplate to code parsing a B tree and trying to check booleans. But I'm fine with that.

Reply:
I was imagining that there's already a `->TRUE` method which is a wrapper of `SvTRUE()` (though, as @tonycoz points out it really should be using `SvTRUE_mg()`. If not, then I'd much prefer adding that one as it's more universally useful - it would apply to any kind of SV then.

Reply:
I'm not particularly a fan of `is_bool_yes` or `is_bool_no`, but more importantly:

B is not the place for abstractions. If an abstract understanding was appropriate, it would be available through normal perl ops. When the internals change, B should reflect that. If you were relying on those internals, your code should break.